### PR TITLE
Add Module#class_eval

### PR DESCRIPTION
**Pull request by mvidner**
Teach Module about the block form of class_eval: https://ruby-doc.org/core-2.5.0/Module.html#method-i-class_eval

Make ` srb tc -e 'String.class_eval { puts self }'` pass.

### Motivation

For example this code needs it: https://github.com/mvidner/ruby-dbus/blob/336f1871bf0f6a0669391fb2bcf7f3a003ec97c9/lib/dbus/proxy_object_interface.rb#L47

### Test plan

I have tried to include tests for this, and they no longer fail, but I still have [doubts. See below.](#issuecomment-505792392)

### CODE REVIEW

File: rbi/core/module.rbi [hunk 1]
```diff
@@ -132,7 +132,13 @@ class Module < Object
     )
     .returns(T.untyped)
   end
-  def class_eval(arg0, filename=T.unsafe(nil), lineno=T.unsafe(nil)); end
+  sig do
+    type_parameters(:U).params(
+        blk: T.proc.params(m: Module).returns(T.type_parameter(:U)),
```
Comment (DarkDimius) [inline rbi/core/module.rbi]: I have minor concern that `Module` type here it too restrictive(it doesn't use type of specific sublcass and thus singleton methods aren't available in scope) and will prohibit most code from being typed without introducing much value.

WDYT about using `T.untyped` to ease adoption?

File: rbi/core/module.rbi [hunk 2]
```diff
@@ -354,7 +360,13 @@ class Module < Object
     )
     .returns(T.untyped)
   end
-  def module_eval(arg0, filename=T.unsafe(nil), lineno=T.unsafe(nil)); end
+  sig do
+    type_parameters(:U).params(
+        blk: T.proc.params(m: Module).returns(T.type_parameter(:U)),
```
Comment (DarkDimius) [inline rbi/core/module.rbi]: ditto

**Comment by mvidner** [comment]
OK, I made the `m` argument untyped, and also realized I should make `self` untyped too.

(It should be fun to figure out the proper type for it, but I must not get distracted into metaprogramming corner cases. Yet.)

**Comment by DarkDimius** [comment]
Thank you, kicked-off CI!

**Comment by DarkDimius** [comment]
@mvidner tests failed, I haven't yet had a chance to look at why.
Would you?

**Comment by mvidner** [comment]
I have made tests pass (checked locally with `bazel test test --config=dbg`), but doing so made my confidence in the code decrease :frowning_face: 

What *is* `light.rbi`?? I've checked that it makes tests pass even if I revert the fix in `rbi/core/module.rbi`. :fire:This:fire:is:fire:fine:fire: I guess?

**Comment by jez** [comment]
@mvidner I removed light.rbi in #1125. I pushed a merge commit to your branch that fixes the merge conflicts that arose because you changed that file, and I verified that your tests still pass locally.

I'll merge this when the tests pass. Thanks for your contribution!

**Comment by mvidner** [comment]
Oh, light is gone, that's great! Thank you!

**Pull request merged**